proto_parser/demo/demo_main.cc

```diff
@@ -0,0 +1,393 @@
+    CopyVal(constructed_field.get(), original);
+
+    return constructed_field;
+  }
+
+  // Original: 
+  //            # Condition 3.1
+  //            leaf_ui_predicate {
+  //              # ...
+  //              ui_element_type: STORE_UI_ELEMENT
+  //              ui_element_matcher {
+  //                input_field: "type"
+  //                int_comparator: 7305  # AUTHENTICATION_OPT_OUT_CHECKED
+  //              }
+  //            }
+  // Modified: 
+  //            # Condition 3.1
+  //            leaf_ui {
+  //              path:  "type"
+  //              equals_int: 7305  # AUTHENTICATION_OPT_OUT_CHECKED
+  //            }
+  std::unique_ptr<ProtoValue> LeafUiPredicateToLeafUi(
+      const ProtoValue& original) {
+    absl::flat_hash_map<std::string, ProtoValue*> field_name_to_value_ptr = 
+        GetFieldNameToPtrMap(*static_cast<const MessageValue*>(&original));
+    return MessageMatcherUniversalConverter(
+        *field_name_to_value_ptr["ui_element_matcher"], "leaf_ui");
+  }
+
+  // Original: 
+  //            # Condition 3.2
+  //            index_ui_matcher {
+  //              ui_element_type: STORE_UI_ELEMENT
+  //              index: -2
+  //              ui_element_matcher {
+  //                input_field: "type"
+  //                int_comparator: 1  # SOMETHING
+  //              }
+  //            }
+  // Modified: 
+  //            # Condition 3.2
+  //            index_ui {
+  //              index: -2
+  //              path: "type"
+  //              equals_int: 1  # SOMETHING
+  //            }
+  std::unique_ptr<ProtoValue> IndexUiPredicateToIndexUi(
+      const ProtoValue& original) {
+    absl::flat_hash_map<std::string, ProtoValue*> field_name_to_value_ptr = 
+        GetFieldNameToPtrMap(*static_cast<const MessageValue*>(&original));
+    return MessageMatcherUniversalConverter(
+        *field_name_to_value_ptr["ui_element_matcher"], "index_ui", 
+        field_name_to_value_ptr["index"]);
+  }
+
+  // Original: 
+  //            # Condition 1
+  //            event_matcher {
+  //              input_field: "PlayExtension.store.click"
+  //            }
+  // Modified:
+  //            # Condition 1
+  //            field_exists: "PlayExtension.store.click"
+  std::unique_ptr<ProtoValue> EventMatcherToFieldExists(
+      const ProtoValue& original) {
+    return MessageMatcherUniversalConverter(original, "field_exists");
+  }
+
+  // Original:
+  //           # Condition 2
+  //           event_matcher {
+  //             input_field: "PlayExtension.log_source"
+  //             int_comparator: 65  # WEB_STORE
+  //           }
+  // Modified:
+  //           # Condition 2
+  //           field_value {
+  //             path: "PlayExtension.log_source"
+  //             equals_int: 65  # WEB_STORE
+  //           }
+  std::unique_ptr<ProtoValue> EventMatcherToFieldValue(
+      const ProtoValue& original) {
+    return MessageMatcherUniversalConverter(original, "field_value");
```

Comment:
I think you can do the implementation here instead of inside MessageMatcherUniversalConverter(), which can help to make functions smaller and more readable?